### PR TITLE
Coverage reporting

### PR DESCRIPTION
**Pull request by psafont**
This adds coverage reporting and uploading to both coveralls and codecov, any of which can be easily disabled.

A common-workflow-language admin should give access to codecov or coveralls to enable uploads to them.

This PR supersedes https://github.com/common-workflow-language/cwltool/pull/526

### CODE REVIEW

**Comment by mr-c** [comment]
Can you also look at the Makefile which has some related targets? It is good to enable local feedback too. Thanks!

**Comment by psafont** [comment]
What is the functionality that is needed in the Makefile?
Generate coverage of all the tests on the default python version and generate a report?

**Comment by mr-c** [comment]
Yes, which might already exist but should be documented in our non-existent developer guide 😋

File: Makefile [hunk 1]
```diff
@@ -113,23 +113,10 @@ pylint_report.txt: ${PYSOURCES}
...
-	coverage combine ${CWL} ${CWL}/draft-3/ ./
+.coverage: tests
 
-coverage.xml: .coverage
-	python-coverage xml
-
-coverage.html: htmlcov/index.html
-
-htmlcov/index.html: .coverage
-	python-coverage html
```
Comment (mr-c) [inline Makefile]: can we keep the HTML report?

Comment (psafont) [reply Makefile]: In retrospect, we can also keep the xml report

**Comment by manu-chroma** [comment]
this is awesome!

**Pull request merged**